.aws/src/main.ts

```diff
@@ -53,38 +51,80 @@ class PrefectV2 extends TerraformStack {
     // Custom construct to implement the IAM roles needed for the agent
     // Abstracting IAM stuff into iam.ts module
 
-    // we need an agent per queue...so we create 2 agent services
-    this.getAgentService('test');
-    this.getAgentService('live');
-
     // create new data-flows-prefect-envs ECR Repo
-    new ApplicationECR(this, 'data-flows-prefect-envs-ecr', {
+    const ecrRepo = new ApplicationECR(this, 'data-flows-prefect-envs-ecr', {
       name: 'data-flows-prefect-envs'
     });
 
-    // create live and test filesystems
-    const testFS = this.createDataFlowsBucket('test');
-    const liveFS = this.createDataFlowsBucket('live');
+    // we need an agent per queue
+    // AWS dev will have the dev agent
+    // AWS production will have the staging and main agent
+    // we also need an s3 bucket per queue and set of iam roles
+    // these map to github branch strategy of dev-v2, staging-v2, and main-v2
+    if (config.isDev) {
+      this.getAgentService('dev');
+      const devS3Bucket = this.createDataFlowsBucket('dev');
+      new DataFlowsIamRoles(
+        this,
+        'dataFlowDevRoles',
+        devS3Bucket,
+        this.caller,
+        this.region,
+        config.tags.environment,
+        'dev',
+        ecrRepo
+      );
+    } else {
+      this.getAgentService('staging');
+      const stagingS3Bucket = this.createDataFlowsBucket('staging');
+      new DataFlowsIamRoles(
+        this,
+        'dataFlowStagingRoles',
+        stagingS3Bucket,
+        this.caller,
+        this.region,
+        config.tags.environment,
+        'staging',
+        ecrRepo
+      );
+      this.getAgentService('main');
+      const mainS3Bucket = this.createDataFlowsBucket('main');
+      new DataFlowsIamRoles(
+        this,
+        'dataFlowMainRoles',
+        mainS3Bucket,
+        this.caller,
+        this.region,
+        config.tags.environment,
+        'main',
+        ecrRepo
+      );
+    }
```

Comment:
this is the creation of deployment type specific resources.

if deployment type is dev, this is going to the AWS dev account and only needs a single:

- s3 bucket
- Prefect agent service
- task and execution roles

else this is going to AWS production and we need a set for `staging` and a set for `main`